src/services/index.ts

```diff
@@ -0,0 +1,448 @@
+      || version === hathorLib.constants.MERGED_MINED_BLOCK_VERSION;
+
+export const handleVertexAccepted = async (context: Context, _event: Event) => {
+  const mysql = await getDbConnection();
+  await mysql.beginTransaction();
+
+  try {
+    const fullNodeEvent = context.event as FullNodeEvent;
+    const now = getUnixTimestamp();
+    const { BLOCK_REWARD_LOCK } = getConfig();
+    const blockRewardLock = BLOCK_REWARD_LOCK;
+
+    // @ts-ignore
+    const {
+      hash,
+      metadata,
+      timestamp,
+      version,
+      weight,
+      outputs,
+      inputs,
+      tokens,
+      token_name,
+      token_symbol,
+    } = fullNodeEvent.event.data;
+
+    const dbTx: DbTransaction | null = await getTransactionById(mysql, hash);
+
+    // Db is already on the database, ignore...
+    if (dbTx) {
+      logger.debug(`Transaction ${hash} already on the database, ignoring..`);
+      mysql.destroy();
+
+      return;
+    }
+
+    let height: number | null = metadata.height;
+
+    if (!isBlock(version) && !metadata.first_block) {
+      height = null;
+    }
+
+    const txOutputs: TxOutputWithIndex[] = prepareOutputs(outputs, tokens);
+    const txInputs: TxInput[] = prepareInputs(inputs, tokens);
+
+    let heightlock = null;
+    if (isBlock(version)) {
+      if (typeof height !== 'number' && !height) {
+        throw new Error('Block with no height set in metadata.');
+      }
+      // unlock older blocks
+      const utxos = await getUtxosLockedAtHeight(mysql, now, height);
+
+      if (utxos.length > 0) {
+        logger.debug(`Block transaction, unlocking ${utxos.length} locked utxos at height ${height}`);
+        await unlockUtxos(mysql, utxos, false);
+      }
+
+      // set heightlock
+      heightlock = height + blockRewardLock;
+
+      // get the first output address
+      const blockRewardOutput = outputs[0];
+
+      // add miner to the miners table
+      // @ts-ignore
+      await addMiner(mysql, blockRewardOutput.decoded.address, hash);
+
+      // here we check if we have any utxos on our database that is locked but
+      // has its timelock < now
+      //
+      // we've decided to do this here considering that it is acceptable to have
+      // a delay between the actual timelock expiration time and the next block
+      // (that will unlock it). This delay is only perceived on the wallet as the
+      // sync mechanism will unlock the timelocked utxos as soon as they are seen
+      // on a received transaction.
+      await unlockTimelockedUtxos(mysql, now);
+    }
+
+    if (version === hathorLib.constants.CREATE_TOKEN_TX_VERSION
+       && token_name
+       && token_symbol) {
+      if (!token_name || !token_symbol) {
+        logger.error('Processed a token creation event but it did not come with token name and symbol');
+        process.exit(1);
+      }
+      await storeTokenInformation(mysql, hash, token_name, token_symbol);
+    }
```

Comment:
Why do we `process.exit(1)` instead of `throw new Error()` like in the block height check above?

Reply:
We should, I'm not sure why I were using process.exit, refactored in f40953f5e5699f530178b8d6868e60dfd6c8589a